aten/src/ATen/WrapDimUtils.h

```diff
@@ -5,16 +5,22 @@
 
 namespace at {
 
-static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr) {
+static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wrap_scalar=true) {
   if (dim_post_expr <= 0) {
```

Comment:
Do we ever have negative dim values?

Reply:
The intent is for `x.size(-1)` to be valid C++ and Python... so, yes?

Reply:
I think @apaszke is probably referring to `tensor.dim()`; it used to be possible to have negative values before we had size {0} tensors, but now they are always non-negative.

`dim_post_expr` is IMO unfortunately named because it's `tensor.dim() + expr`, i.e. has nothing really to do with the `dim` parameter and can't currently be negative (because expr is always 0 or 1 and dim is always at least 0).

I'm open to suggestions for making this clearer -- maybe rename the parameter `ndim_post_expr` and just check == 0?

Reply:
where `expr` is... whatever expression you use to select the dimension? So we add `tensor.dim()` even to positive values? That's confusing

Reply:
No, it's the thing to handle that the range allowed can be bigger than the number of dimensions, e.g. unsqueeze you can unsqueeze past the current end.